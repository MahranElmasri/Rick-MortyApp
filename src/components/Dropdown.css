.dropdown {
  display: inline;
  margin: 20px 10px;
}
