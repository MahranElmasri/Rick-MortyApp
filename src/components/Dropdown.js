import React, { useState } from "react";
import "./Dropdown.css";
import PropTypes from "prop-types";

export default function Dropdown({ onSelect, options, placeholder, filter }) {
  const [value, setValue] = useState("");
  const onChange = (e) => {
    e.preventDefault();
    // setValue();
    onSelect(e.target.value, filter);
  };

  return (
    <div className="dropdown">
      <select data-testid="dropdown" onChange={onChange}>
        <option value="">{placeholder}</option>
        {options.map((value) => (
          <option key={value} value={value}>
            {value}
          </option>
        ))}
      </select>
    </div>
  );
}

Dropdown.propTypes = {
  options: PropTypes.array.isRequired,
  onSelect: PropTypes.func.isRequired,
  placeholder: PropTypes.string.isRequired,
};
