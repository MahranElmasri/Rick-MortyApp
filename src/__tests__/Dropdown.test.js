import Dropdown from "../components/Dropdown";

it("should render correctlly", () => {
  const options = [];
  const onChange = jest.fn();
  const { getByTestId } = render(
    <Dropdown
      options={options}
      onChange={onChange}
      onSelect={jest.fn()}
      placeholder="Filter by status"
    />
  );

  expect(getByTestId("dropdown")).toMatchSnapshot();
});

it("should render placeholder text", () => {
  const options = [];
  const onChange = jest.fn();
  const { getByTestId, getByText } = render(
    <Dropdown
      options={options}
      onChange={onChange}
      onSelect={jest.fn()}
      placeholder="Filter by status"
    />
  );
  expect(getByTestId("dropdown")).toHaveTextContent("Filter by status");
});

it("should render first select text", () => {
  const options = ["Alive", "Dead", "Unknown"];
  const onChange = jest.fn();
  const { getByTestId, getByText } = render(
    <Dropdown
      options={options}
      onChange={onChange}
      onSelect={jest.fn()}
      placeholder="Filter by status"
    />
  );
  expect(getByTestId("dropdown")).toHaveTextContent("Filter by status");
  fireEvent.click(getByText("Alive"));
  expect(getByTestId("dropdown")).toHaveTextContent("Alive");
});
