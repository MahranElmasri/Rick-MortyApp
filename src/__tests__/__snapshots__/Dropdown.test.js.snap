// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctlly 1`] = `
<select
  data-testid="dropdown"
>
  <option
    value=""
  >
    Filter by status
  </option>
</select>
`;
